### PR TITLE
#167891568 Integrate Coveralls code coverage service

### PR DESCRIPTION
**Pull request by Pomile**
#### What does this PR do?
* Have code coverage service integration
#### Description of Task to be completed?
- Install coveralls, cross-env, babel-plugin-instanbul
- Configure nyc not to instrument code
- Configure travis.yml
- Configure babelrc
- Modify test script in package.json

#### How should this be manually tested?
Run `npm install`
Run `npm test`
#### What are the relevant pivotal tracker stories?
[#167891568](https://www.pivotaltracker.com/story/show/167891568)
#### Any background context you want to add?
Nil
#### Screenshots
Nil

### CODE REVIEW

**Review by tvpeter — APPROVED** [review]
Nice job. However, do indicate that users should run `npm install` before running `npm test`. Thank you.

**Review by muheebolakunle — COMMENTED** [review]
Good job, I think you should add the test folder to the **exclude** property of **nyc** in the package.json to avoid covering the test files and skewing our coverage result. Thank you

**Review by thatkazuk1 — DISMISSED** [review]
Great job.

This should work also:
```
`"test": "cross-env-shell NODE_ENV=test nyc mocha -r @babel/register -r @babel/polyfill --exit"`
```

**Comment by Pomile** [comment]
> Nice job. However, do indicate that users should run `npm install` before running `npm test`. Thank you.

Done. Thank you Peter.

**Comment by Pomile** [comment]
> Nice job. However, do indicate that users should run `npm install` before running `npm test`. Thank you.

Done. Thank you Peter

**Comment by Pomile** [comment]
> Good job, I think you should add the test folder to the **exclude** property of **nyc** in the package.json to avoid covering the test files and skewing our coverage result. Thank you

Thank you Muheeb for your feedback.

This has been done already. I configured **nyc**, not to instrument code which also include test folder. **nyc** would only report the `/src` folder excluding the subfolders that were stated in the **exclude** array.

**Comment by Pomile** [comment]
**Why 3 test script:**
1.	`mocha:test`: uses mocha to run test file(s). This is used in the `test` script.
2.	`test`: uses cross-env to set NODE_ENV to test environment, report with `nyc`  and run `mocha:test`.
3.	`local:test`: uses nodemon to watch the sources folder for changes and executes `npm test`. This is necessary in order to save time in development.

**Review by thatkazuk1 — APPROVED** [review]
Nice job. Now, we can move forward as the Lord intended😁

File: package.json [hunk 1]
```diff
@@ -11,7 +25,8 @@
     "build": "npm run clean && npm run build-babel",
     "devstart": "nodemon --exec babel-node src/index.js",
     "lint": "eslint --fix ./src/**/*.js",
-    "test": "nyc mocha --require @babel/register --require @babel/polyfill --timeout 1000 --exit"
+    "test": "cross-env NODE_ENV=test nyc --reporter=lcov --reporter=text mocha --require @babel/register --require @babel/polyfill --exit",
```
Comment (daniellamarr) [inline package.json]: - What's your reason for having `cross-env`?
- I don't think you should have this `--reporter=lcov --reporter=text` in your `test script`. Only coverage should have it

**Review by daniellamarr — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Pomile) [reply package.json]: `nyc` collect all the coverage report. 
`babel-plugin-istanbul` respects the include/exclude configuration options from `nyc`, but you also need to configure NYC not to instrument your code by adding these settings in your package.json: 
I need to swtich environment to test for nyc include / exclude to work. 
see [Babel-plugin-instanbul](https://github.com/istanbuljs/babel-plugin-istanbul) for more details.
I will remove ` --reporter=lcov --reporter=text `.

**Review by max-wel — COMMENTED** [review]
Good job

File: package.json (hunk 1 above)
Comment (Pomile) [reply package.json]: Done. I have removed `--reporter=lcov --reporter=text` from test script.

File: package.json [hunk 2]
```diff
@@ -11,7 +25,8 @@
     "build": "npm run clean && npm run build-babel",
     "devstart": "nodemon --exec babel-node src/index.js",
     "lint": "eslint --fix ./src/**/*.js",
-    "test": "nyc mocha --require @babel/register --require @babel/polyfill --timeout 1000 --exit"
+    "test": "cross-env NODE_ENV=test nyc mocha --require @babel/register --require @babel/polyfill --exit",
```
Comment (daniellamarr) [inline package.json:28]: I just tried testing without cross-env, but I didn't experience any issues

**Review by daniellamarr — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Pomile) [reply package.json:28]: It was not working from my own end when i ran `npm test` without cross-env. I reach out to @meetKazuki in this regard and the result is still the same.
Please advice

Comment (Pomile) [reply package.json:28]: error

Comment (max-wel) [reply package.json:28]: Try running it without the ```NODE_ENV=test``` prefix

**Review by daniellamarr — APPROVED** [review]

**Pull request merged**